pypeit/core/wavecal/wvutils.py

```diff
@@ -552,7 +555,10 @@ def xcorr_shift(inspec1, inspec2, percent_ceil=50.0, use_raw_arc=False, sigdetec
         return 0.0, 0.0
 
     nspec = y1.shape[0]
-    lags = np.arange(-nspec + 1, nspec)
+    if lag_range is None:
+        lags = np.arange(-nspec + 1, nspec)
+    else:
+        lags = np.linspace(lag_range[0], lag_range[1], 2*nspec-1)
```

Comment:
Is it important the number of lags searched in each case is the same?  I.e., is that why the number of samples here is `2*nspec-1`?

Reply:
Yes, exactly.  Later in the code there is an interpolation that puts constrains on the shape of lags.